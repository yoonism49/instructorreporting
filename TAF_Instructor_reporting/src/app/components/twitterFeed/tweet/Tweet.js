// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';

const Tweet = ({
  time,
  author,
  authorAvatar,
  children
}) => (
  <li className="media">
    <a href="#" className="pull-left">
      <img
        src={authorAvatar}
        alt="Avatar"
        className="img-circle"
        width="64"
        height="64"
      />
    </a>
    <div className="media-body">
      <span className="text-muted pull-right">
        <small>
          <em>
            {time}
          </em>
        </small>
      </span>
      <a href="#">
        <strong>
          {author}
        </strong>
      </a>
      { children }
    </div>
  </li>
);

Tweet.propTypes = {
  time:         PropTypes.string.isRequired,
  author:       PropTypes.string.isRequired,
  authorAvatar: PropTypes.any.isRequired,
  children:     PropTypes.node.isRequired
};

export default Tweet;
