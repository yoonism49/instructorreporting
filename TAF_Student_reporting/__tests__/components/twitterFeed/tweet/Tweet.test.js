'use strict';

import React                from 'react';
import Tweet                from '../../../../src/app/components/twitterFeed/tweet/Tweet';
import renderer             from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('Tweet component', () => {
  it('renders as expected', () => {
    const mockProps = {
      time: '12:21',
      author: 'Mackentoch',
      authorAvatar: ''
    };
    const component = renderer.create(
      <div>
        <Tweet {...mockProps}>
          <p>Tweet content</p>
        </Tweet>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
