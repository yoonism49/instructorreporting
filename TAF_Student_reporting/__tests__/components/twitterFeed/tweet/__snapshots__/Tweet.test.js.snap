// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet component renders as expected 1`] = `
<div>
  <li
    className="media"
  >
    <a
      className="pull-left"
      href="#"
    >
      <img
        alt="Avatar"
        className="img-circle"
        height="64"
        src=""
        width="64"
      />
    </a>
    <div
      className="media-body"
    >
      <span
        className="text-muted pull-right"
      >
        <small>
          <em>
            12:21
          </em>
        </small>
      </span>
      <a
        href="#"
      >
        <strong>
          Mackentoch
        </strong>
      </a>
      <p>
        Tweet content
      </p>
    </div>
  </li>
</div>
`;
